File: web/src/models/DTOs/ConvenioBancario/ConvenioBancarioDTO.ts
import { MoedaDTO } from ".";
import { ContaCorrenteDTO } from "../AgenciaBancoConta/ContaCorrenteDTO";
import { UnidadeDTO } from "../Outros/UnidadeDTO";
import { FaixaNossoNumeroDTO } from "./FaixaNossoNumeroDTO";

export type CarteiraConvenioBancario = "RG" | "SR";
enum CarteiraConvenioBancarioEnum {
  RG = "RG",
  SR = "SR",
}
export const carteiraConvenioBancarioValues = Object.values(CarteiraConvenioBancarioEnum);

export type SistemaBancario = "SINCO" | "SIGCB";
enum SistemaBancarioEnum {
  SINCO = "SINCO",
  SIGCB = "SIGCB",
}
export const sistemaBancarioValues = Object.values(SistemaBancarioEnum);

export interface ConvenioBancarioDTO {
  id: number;
  contaCorrente: ContaCorrenteDTO;
  nomeCedente: string;
  numero: string;
  carteira: CarteiraConvenioBancario;
  moeda: MoedaDTO;
  tituloEspecie: string;
  tipoEmissao: string;
  aceite: boolean;
  indiceMulta: number;
  indiceJuros: number;
  sistemaBancario: SistemaBancario;
  observacao1: string | null;
  observacao2: string | null;
  observacao3: string | null;
  observacao4: string | null;
  observacao5: string | null;
  localPagamento: string;
  idUnidade: number;
  dataInativo: Date | null;
  dataInclusao: Date;
  idOperadorInclusao: number;
  dataAlteracao: Date | null;
  idOperadorAlteracao: number | null;
  utilizaUnCentralizadora: boolean;
  faixasNossoNumero: FaixaNossoNumeroDTO[];

  unidade: UnidadeDTO;
}

File: webservices/src/main/java/fieg/modulos/cadastro/conveniobancario/dto/ConvenioBancarioDTO.java
package fieg.modulos.cadastro.conveniobancario.dto;

import fieg.modulos.cadastro.contacorrente.dto.ContaCorrenteDTO;
import fieg.modulos.cadastro.conveniobancario.enums.SistemaBancario;
import fieg.modulos.cadastro.conveniobancario.enums.CarteiraConvenioBancario;
import fieg.modulos.unidade.dto.UnidadeDTO;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@ToString
public class ConvenioBancarioDTO {

    private Integer id;
    private ContaCorrenteDTO contaCorrente;
    private String nomeCedente;
    private String numero;
    private CarteiraConvenioBancario carteira;
    private MoedaDTO moeda;
    private String tituloEspecie;
    private String tipoEmissao;
    private Boolean aceite;
    private Float indiceMulta;
    private Float indiceJuros;
    private SistemaBancario sistemaBancario;
    private String observacao1;
    private String observacao2;
    private String observacao3;
    private String observacao4;
    private String observacao5;
    private String localPagamento;
    private Integer idUnidade;
    private LocalDateTime dataInativo;
    private LocalDateTime dataInclusao;
    private Integer idOperadorInclusao;
    private LocalDateTime dataAlteracao;
    private Integer idOperadorAlteracao;
    private Boolean utilizaUnCentralizadora;
    private List<FaixaNossoNumeroDTO> faixasNossoNumero;

    private UnidadeDTO unidade;
}
